### PR TITLE
Tidy Test Playbook

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35969

## Description
Fixed some small bugs in Tidy integration: context path, error handling and support for SSH key to use on test playbook.
Added Tidy test playbook

## Minimum version of Cortex XSOAR
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] No

### CODE REVIEW

File: Packs/Tidy/Integrations/Tidy/Tidy.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ configuration:
   name: password
   required: true
   type: 4
+- display: SSH Key
+  hidden: false
+  name: ssh_key
+  required: false
+  type: 9
```
Comment (Itay4) [inline Packs/Tidy/Integrations/Tidy/Tidy.yml:25]:
```suggestion
  type: 15
```

File: Packs/Tidy/Integrations/Tidy/Tidy.py [hunk 2]
```diff
@@ -82,12 +83,19 @@ def _execute(self, playbook_name: str, extra_vars=None) -> Runner:
         """
         if extra_vars is None:
             extra_vars = {}
+        inventory = f"{self.username}@{self.hostname} ansible_host=\"{self.hostname}\" " \
+                    f"ansible_user=\"{self.username}\" ansible_password=\"{self.password}\" " \
+                    f"ansible_become_password=\"{self.password}\" ansible_connection=ssh"
+        if self.ssh_key:
+            with open('key.pem', "w") as ssh_key_file:
```
Comment (Itay4) [inline Packs/Tidy/Integrations/Tidy/Tidy.py:90]:
```suggestion
            with open('key.pem', 'w') as ssh_key_file:
```

File: Tests/conf.json [hunk 3]
```diff
@@ -3288,6 +3291,7 @@
         "Phishing Classifier V2 ML Test": "Issue 26066",
         "RedLockTest": "Issue 24600",
         "SentinelOne V2 - test": "Issue 24933",
+        "Tidy - Test": "Will run manually once a month",
```
Comment (Itay4) [inline Tests/conf.json:3294]: lets not skip it immediately, but let it run for a few days

File: Tests/conf.json [hunk 4]
```diff
@@ -3277,6 +3277,9 @@
             "integrations": "Armis",
             "playbookID": "Armis-Test",
             "fromversion": "5.5.0"
+        },
+        {
+            "playbookID": "Tidy - Test"
```
Comment (Itay4) [inline Tests/conf.json:3282]: shouldnt you configure the integrations here as well?

File: Packs/Tidy/Integrations/Tidy/Tidy.py [hunk 5]
```diff
@@ -653,7 +664,11 @@ def main() -> None:
     hostname = demisto.getArg("hostname") or demisto.getParam("hostname")
     user = demisto.getArg("user") or demisto.getParam("user")
     password = demisto.getArg("password") or demisto.getParam("password")
-    client = TidyClient(hostname=hostname, user=user, password=password)
+    ssh_key = demisto.getArg("ssh_key") or demisto.getParam("ssh_key")
+    if ssh_key:
+        client = TidyClient(hostname=hostname, user=user, password=password, ssh_key=ssh_key)
```
Comment (Itay4) [inline Packs/Tidy/Integrations/Tidy/Tidy.py:669]: lets not call `TidyClient` in the condition, but build the args and then initialize the client after the condition

File: Packs/Tidy/Integrations/Tidy/Tidy.yml [hunk 6]
```diff
@@ -18,6 +18,11 @@ configuration:
   name: password
   required: true
   type: 4
+- display: SSH Key
```
Comment (Itay4) [inline Packs/Tidy/Integrations/Tidy/Tidy.yml:21]: please add tooltip what this ssh key used for

File: Packs/Tidy/Integrations/Tidy/Tidy.py [hunk 7]
```diff
@@ -653,7 +664,11 @@ def main() -> None:
     hostname = demisto.getArg("hostname") or demisto.getParam("hostname")
     user = demisto.getArg("user") or demisto.getParam("user")
     password = demisto.getArg("password") or demisto.getParam("password")
-    client = TidyClient(hostname=hostname, user=user, password=password)
+    ssh_key = demisto.getArg("ssh_key") or demisto.getParam("ssh_key")
```
Comment (Itay4) [inline Packs/Tidy/Integrations/Tidy/Tidy.py:667]: why are we doing `getArg` ?

**Review by Itay4 — COMMENTED** [review]
looks good
1. please make sure we use `CommandResults`
2. please make sure the README is up to our standards
3. in the test playbook, did we think of a way to verify that the content developer setup playbook ran successfully?

File: Packs/Tidy/Integrations/Tidy/Tidy.yml (hunk 1 above)
Comment (Shellyber) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: Why 15? it's for single select

Comment (Itay4) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: 14*

Comment (Shellyber) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: I think it should stay 9. 
It shouldn't be visible. We're using a pem key here.

Comment (Itay4) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: 14 is long text encrypted (not visible) 
why do we need authentication param for ssh key?

Comment (Shellyber) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: The pycharm plugin showed me it's a short text so I thought it's regular. Shouldn't it be 12? 
I think 14 is for short encrypted and 12 is for long (checked in with teams integration).

Comment (Itay4) [reply Packs/Tidy/Integrations/Tidy/Tidy.yml:25]: 12 is long text (not encrypted), 14 is long text encrypted

**Review by Itay4 — APPROVED** [review]

**Comment by Shellyber** [comment]
@roysagi Please review the rn

**Pull request merged**